packages/blade/src/components/Input/_decisions/decisions.md

```diff
@@ -0,0 +1,295 @@
+# Input Decisions <!-- omit in toc -->
+
+This doc will talk about the API decisions for all the types of input elements
+
+- [x] TextInput,
+- [x] PasswordInput,
+- [ ] OTPInput,
+- [ ] CardInput,
+- [ ] TextArea
+
+## Index <!-- omit in toc -->
+
+- [Design](#design)
+- [BaseInput API](#baseinput-api)
+  - [The blueprint layout of icons, prefix, suffix and loader](#the-blueprint-layout-of-icons-prefix-suffix-and-loader)
+- [TextInput API](#textinput-api)
+  - [Web & React Native Attributes for showing keyboard types and autocomplete suggestions based on input](#web--react-native-attributes-for-showing-keyboard-types-and-autocomplete-suggestions-based-on-input)
+  - [Return Key types for web and Native](#return-key-types-for-web-and-native)
+- [PasswordInput](#passwordinput)
+- [Notes](#notes)
+- [Open questions](#open-questions)
+- [Discussion Notes](#discussion-notes)
+  - [Actionables for Design based on discussions on 21-07-2022](#actionables-for-design-based-on-discussions-on-21-07-2022)
+- [References](#references)
+
+## Design
+
+- [TextInput](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=10953%3A191574)
+- [PasswordInput](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=10953%3A180381)
+- [OTPInput]()
+- [CardInput]()
+- [TextArea]()
+
+## BaseInput API
+
+The BaseInput is the underlying input field that will in turn be used to render all types of input. For now this will be an **internal component** and won't be exposed to consumers. We need this so we can have the rendering of text input and interactions at one central place but can restrict/add more control to consumer components based on their type.
+
+| Prop                                                                                              | Type                                                                                                                                                                                                                                                   | Required | Default                                                   | Description                                                                                                                                                                                                                                                                           |
```

Comment:
>autoCapitalize
What is the use case you have seen for this?

>onSubmitEditiong
what is the use case for this? people adding information and hitting the submit/return button on the keyboard and the form shall submit?

>maxLength
same as maxCharacters

>multiline
will evaluate once we reach text area. this is right now scoped to TextInput and PasswordInput

>editable
covered as `readOnly`

>placeholderTextColor
why do we need this? ideally this would be automatically handled by the input field based on the other colors. it's the basic styling that isn't configurable by the consumer.